azext_edge/edge/providers/check/deviceregistry.py

```diff
@@ -0,0 +1,612 @@
+                    data_point_data_source_status = CheckTaskStatus.success.value
+                    if data_point_data_source:
+                        data_point_data_source_text = (
+                            f"- Data source: {{[bright_blue]{data_point_data_source}[/bright_blue]}} [green]detected[/green]."
+                        )
+                    else:
+                        data_point_data_source_text = (
+                            "Data source [red]not detected[/red]."
+                        )
+                        data_point_data_source_status = CheckTaskStatus.error.value
+
+                    add_display_and_eval(
+                        check_manager=check_manager,
+                        target_name=target_assets,
+                        display_text=data_point_data_source_text,
+                        eval_status=data_point_data_source_status,
+                        eval_value=data_point_data_source_value,
+                        resource_name=asset_name,
+                        namespace=namespace,
+                        padding=(0, 0, 0, datapoint_padding)
+                    )
+
+                    if detail_level > ResourceOutputDetailLevel.summary.value:
+                        process_properties(
+                            check_manager=check_manager,
+                            detail_level=detail_level,
+                            target_name=target_assets,
+                            prop_value=data_point,
+                            properties=ASSET_DATAPOINT_PROPERTIES,
+                            namespace=namespace,
+                            padding=(0, 0, 0, datapoint_padding + PADDING_SIZE)
+                        )
+
+            if detail_level > ResourceOutputDetailLevel.summary.value:
+                process_properties(
+                    check_manager=check_manager,
+                    detail_level=detail_level,
+                    target_name=target_assets,
+                    prop_value=asset_spec,
+                    properties=ASSET_PROPERTIES,
+                    namespace=namespace,
+                    padding=(0, 0, 0, spec_padding)
+                )
+
+            # events
+            events = asset_spec.get("events", [])
+            if events:
+                check_manager.add_target_conditions(
+                    target_name=target_assets,
+                    namespace=namespace,
+                    conditions=["len(spec.events)"]
+                )
+                events_count = len(events)
+                events_count_value = {"len(spec.events)": events_count}
+                events_count_status = CheckTaskStatus.success.value
+
+                if events_count > 1000:
+                    events_count_text = (
+                        # expecting no more than 1000 events per asset
+                        f"Events [red]exceeding 1000[/red]. Detected {events_count}."
```

Comment:
Curious - is this a hard requirement from the service / do we expect this value of `1000` to change?

Reply:
Hi Ryan, I get this requirement from their CRD definition [here](https://msazure.visualstudio.com/One/_git/Azure-IoT-Platform-DeviceRegistry?path=/arcExtension/common/df/helm/crds/deviceregistry.microsoft.com_assets.yaml&version=GBmain&line=149&lineEnd=149&lineStartColumn=17&lineEndColumn=31&lineStyle=plain&_a=contents), therefore it should be a service hard requirement

Reply:
Ok, sounds good - maybe we should pull this into a constant? Just in case the template changes it may be hard for us to track down this value

Reply:
sounds good! will update